### DISCUSSION TITLE
Getting different plots for monthly seasonality from model.plot_components and  by plotting it from component breakdown

### DISCUSSION THREAD

**Discussion by deepanshu650** [ourownstory/neural_prophet#1224] [Q&A - forecasting best practices] [answered]
Hi all,

I am getting 2 different plots for added monthly seasonality from model.plot_components and once plotting on own from seasonality component breakdown on predicted dataframe.

<img width="720" alt="image" src="https://user-images.githubusercontent.com/65191985/225992748-30e3ba26-04d7-4222-b357-a0b0977771e8.png">

This one i got from model.plot_components

<img width="736" alt="image" src="https://user-images.githubusercontent.com/65191985/225992881-7b4ad2b5-e7bc-45bd-998d-dcade60d8cc4.png">

This one i plotted using monthly seasonality component
Seems like the second one which i got from prediction component is correct as seasonality follows this structure and i am unable to figure out why model.plot_components gave some sort of shifted plot

**Comment by LeonieFreisinger** [comment, marked as answer]
@deepanshu650 The` plot_components()` is a custom neuralprophet function. Please have a look at the docu to checkout its purpose: https://neuralprophet.com/code/forecaster.html?highlight=plot_components#neuralprophet.forecaster.NeuralProphet.plot_components

**Reply by deepanshu650** [reply]
@LeonieFreisinger thanks for your reply,

After going through the code of the function `plot_parameters()` , I see that for plotting custom seasonality (using function `plot_custom_season`), first it creates evenly spaced values (n_steps = 300 values) between 0 to 1 and then for each value it calculates the corresponding fourier series considering it as the time stamp.

After this it plots curve of these evenly spaced values with their corresponding fourier series.
Since it plots with x-axis as values from 0 to 1, this creates a different curve then expected (bit shifted in x-axis) as I mentioned above.

I also checked for Prophet, their the curve generated for custom seasonality lets say monthly seasonality, is with x-axis as days of month.
Their they are using some fix start date (2017-01-01) and then calculate an end date =start date +  period of seasoanlity, and then sample some points in between these dates and caluclate corresponding fourier series. 

If my understanding is correct, can we change the implementation of plotting in this manner.
